### PR TITLE
gh-82616: Add Py_IS_TYPE_SIGNED() macro

### PR DESCRIPTION
**Pull request by vstinner**
_posixsubprocess: add a static assertion to ensure that the pid_t
type is signed.

Replace _Py_IntegralTypeSigned() with Py_IS_TYPE_SIGNED().

### CODE REVIEW

**Comment by vstinner** [comment]
cc @serhiy-storchaka @erlend-aasland @corona10

**Comment by serhiy-storchaka** [comment]
I did not know about `_Py_IntegralTypeSigned()`. It could be used in Modules/posixmodule.c and other interfaces to OS.

But I hesitate about adding it to the public API. Compilers can complain about "always true/false" condition. And what to do with this in PyPy and bindings in other programming languages?

**Comment by serhiy-storchaka** [comment]
I did not know about `_Py_IntegralTypeSigned()`. It could be used in Modules/posixmodule.c and other interfaces to OS.

But I hesitate about adding it to the public API. Compilers can complain about "always true/false" condition. And what to do with this in PyPy and bindings in other programming languages?

**Comment by vstinner** [comment]
Ok, I add it as a private macro instead.

**Comment by vstinner** [comment]
Oh, the Ubtuntu job logs a compiler warning:

comparison of unsigned expression < 0 is always false [-Wtype-limits]

**Review by gpshead — APPROVED** [review]

**Review by erlend-aasland — APPROVED** [review]
Neat trick to get rid of the compiler warning! LGTM, but I'd run it through the buildbots one more time before merging.

**Comment by vstinner** [comment]
> buildbot/PPC64 Fedora PR — Build done. 

Unrelated error: ../configure: line 2692: cannot create temp file for here-document: No space left on device

**Comment by vstinner** [comment]
Merged, thanks for the reviews.

**Pull request merged**
